tests/testapp/home/decorators.py

```diff
@@ -0,0 +1,27 @@
+from draftjs_exporter.dom import DOM
+from wagtail.wagtailcore.models import Page
+
+MISSING_RESOURCE_CLASS = 'button--missing-resource'
+MISSING_RESOURCE_URL = '#'
+
+
+class Button:
+    def render(self, props):
+        link_type = props.get('linkType', '')
+
+        if link_type == 'page':
+            try:
+                page_id = props.get('id')
+                page = Page.objects.get(id=page_id)
+                href = page.url
+            except Page.DoesNotExist:
+                href = props.get('url', MISSING_RESOURCE_URL)
+        else:
+            href = props.get('url', MISSING_RESOURCE_URL)
+
+        anchor_properties = {
+            'className': 'button',
+            'onclick': "window.location.href='{}'".format(href),
```

Comment:
I was struggling to come up with an element which would reuse an existing `source` and still do something interesting, but I'll just convert this to an anchor with a regular href, explaining this is just for demo purposes.